homeassistant/components/actiontec/device_tracker.py

```diff
@@ -31,7 +31,9 @@
 )
 
 
-def get_scanner(hass: HomeAssistant, config: ConfigType) -> DeviceScanner | None:
+def get_scanner(
+    hass: HomeAssistant, config: ConfigType
+) -> ActiontecDeviceScanner | None:
```

Comment:
It is correct, but I don't know if it's necessary / if we should do it. `DeviceScanner | None` is enough, the this change could result in slightly more work when renaming the class. Although that is more theoretical.

Since you've already done the work of creating this PR, I'm inclined to accept it. I would just recommend not change any more that aren't necessary in the future. Do I miss anything?

Reply:
I can split the PR (components versus pylint plugin) if you wish, but it is just part of the general discussion to make return types as precise as possible.

Reply:
It's fine. I just think it isn't worth the effort to update the return types in these cases. We don't gain anything from it. IMO it would be ok to leave them as is.